src/callbacks_step/summary.jl

```diff
@@ -14,42 +22,160 @@ function Base.show(io::IO, cb::DiscreteCallback{Condition,Affect!}) where {Condi
+  println(io, "│ " * "═"^length(heading) * " "^(total_width - length(heading) - 4) * " │")
+end
+
+function summary_line(io, key, value; key_width=30, total_width=100, indentation_level=0)
+  key_width = get(io, :key_width, key_width)
+  total_width = get(io, :total_width, total_width)
+  indentation_level = get(io, :indentation_level, indentation_level)
+
+  s = format_key_value_line(key, value, key_width, total_width,
+                            indentation_level=indentation_level)
+
+  println(io, s)
+end
+
+function summary_footer(io; total_width=100, indentation_level=0)
+  total_width = get(io, :total_width, 100)
+  indentation_level = get(io, :indentation_level, 0)
+
+  if indentation_level == 0
+    s = "└" * "─"^(total_width-2) * "┘"
+  else
+    s = ""
+  end
+
+  print(io, s)
+end
+
+@inline increment_indent(io) = IOContext(io, :indentation_level => get(io, :indentation_level, 0) + 1)
+
+
+# Print information about the current simulation setup
+# Note: This is called *after* all initialization is done, but *before* the first time step
 function initialize_summary_callback(cb::DiscreteCallback, u, t, integrator)
 
   mpi_isroot() || return nothing
 
   print_startup_message()
 
   io = stdout
+  io_context = IOContext(io,
+                         :compact => false,
+                         :key_width => 30,
+                         :total_width => 100,
+                         :indentation_level => 0)
+
   semi = integrator.p
-  show(io, MIME"text/plain"(), semi)
+  show(io_context, MIME"text/plain"(), semi)
   println(io, "\n")
   mesh, equations, solver, _ = mesh_equations_solver_cache(semi)
-  show(io, MIME"text/plain"(), mesh)
+  show(io_context, MIME"text/plain"(), mesh)
   println(io, "\n")
-  show(io, MIME"text/plain"(), equations)
+  show(io_context, MIME"text/plain"(), equations)
   println(io, "\n")
-  show(io, MIME"text/plain"(), solver)
+  show(io_context, MIME"text/plain"(), solver)
   println(io, "\n")
 
   callbacks = integrator.opts.callback
   if callbacks isa CallbackSet
     for cb in callbacks.continuous_callbacks
-      show(io, MIME"text/plain"(), cb)
+      show(io_context, MIME"text/plain"(), cb)
       println(io, "\n")
     end
     for cb in callbacks.discrete_callbacks
-      show(io, MIME"text/plain"(), cb)
+      # Do not show ourselves
+      cb.affect! === summary_callback && continue
+
+      show(io_context, MIME"text/plain"(), cb)
       println(io, "\n")
     end
   else
-    show(io, MIME"text/plain"(), callbacks)
+    show(io_context, MIME"text/plain"(), callbacks)
     println(io, "\n")
   end
 
-  # TODO: Taal decide, shall we print more information about the ODE problem/algorithm?
-  println(io, "tspan = ", integrator.sol.prob.tspan)
-  println("Time integrator: ", integrator.alg)
+  setup = Pair{String,Any}[
+           "Start time" => first(integrator.sol.prob.tspan),
+           "Final time" => last(integrator.sol.prob.tspan),
+           "time integrator" => typeof(integrator.alg).name,
```

Comment:
This will remove any information about limiters (stage callbacks). Is that okay/desired?

Reply:
Not really, since we shouldn't commit type piracy. The output should be improved in OrdinaryDiffEq.jl (where this algorithm is defined). You can also keep your current version.

Reply:
OK, then I'll do that.